### PR TITLE
ENH: allow for custom montage in config file

### PR DESCRIPTION
**Pull request by dengemann**
let's see how it goes -- added it to one of the ERP core configs @agramfort @hoechenberger 

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -2547,13 +2547,17 @@ def _set_eeg_montage(cfg, raw, subject, session) -> None:
 
     Modifies ``raw`` in-place.
     """
-    montage_name = cfg.eeg_template_montage
-    if cfg.datatype == 'eeg' and montage_name:
+    montage = cfg.eeg_template_montage
+    is_mne_montage = isinstance(montage,
+                                mne.channels.montage.DigMontage)
+    montage_name = ('custom_montage' if is_mne_montage else montage)
```
Comment (hoechenberger) [inline config.py:2553]:
```suggestion
    montage_name = 'custom_montage' if is_mne_montage else montage
```

**Review by hoechenberger — COMMENTED** [review]
Looks good to me, let's wait for CIs to come back green

@dengemann Could you add a changelog entry please?

File: config.py (hunk 1 above)
Comment (dengemann) [reply config.py:2553]: It can help readability to have the parenthesis, often done in MNE code.

Comment (hoechenberger) [reply config.py:2553]: I've honestly never seen this in MNE 🤔 
If you're concerned about readability, shouldn't we do
```python
if is_mne_montage:
    montage_name = 'custom_montage'
else:
    montage_name = montage
```
?

Comment (dengemann) [reply config.py:2553]: do as you prefer, I don't care very much. I find my one-line a good compromise though :)

**Comment by dengemann** [comment]
Cool ERP_CORE seems to have worked

**Comment by dengemann** [comment]
the ds248 seems unrelated.

**Comment by hoechenberger** [comment]
> the ds248 seems unrelated.

Yes, sometimes we get a timeout even after a successful run, as happened here

**Comment by dengemann** [comment]
OK to avoid debate, I committed your suggestion. From my side good to go

**Review by agramfort — COMMENTED** [review]
LGTM

can you update changelog?

**Comment by dengemann** [comment]
pushed changes.md @hoechenberger @agramfort

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
Ci complain 

no objection to merge when CIs are green

**Comment by dengemann** [comment]
```
18:50:09 Using custom configuration: tests/configs/config_ds003392.py
18:50:09 Using custom configuration: tests/configs/config_ds000246.py
18:50:09 Using custom configuration: tests/configs/config_ds000247.py
18:50:09 Using custom configuration: tests/configs/config_ds000248.py
18:50:09 Using custom configuration: tests/configs/config_ds000248_ica.py
18:50:09 Using custom configuration: tests/configs/config_ds000248_no_mri.py
18:50:09 Using custom configuration: tests/configs/config_ds001810.py
18:50:09 Using custom configuration: tests/configs/config_eeg_matchingpennies.py
18:50:09 Using custom configuration: tests/configs/config_ds003104.py
18:50:09 Using custom configuration: tests/configs/config_ds000117.py
18:50:09 Using custom configuration: tests/configs/config_ERP_CORE.py
Traceback (most recent call last):
  File "docs/source/examples/gen_examples.py", line 179, in <module>
    demonstrated_funcs_str = gen_demonstrated_funcs_str(config_path)
  File "docs/source/examples/gen_examples.py", line 30, in gen_demonstrated_funcs_str
    config = runpy.run_path(config_module_path)
  File "/usr/local/lib/python3.8/runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/usr/local/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/local/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "config.py", line 2797, in <module>
    if (get_task() is not None and
  File "config.py", line 1938, in get_task
    raise ValueError(f'Invalid task. It can be: '
ValueError: Invalid task. It can be:  but got: N400

Exited with code exit status 1
```

No idea why now out of a sudden the CI would do this. @hoechenberger ? My change in last commit was unrelated and it used to work.

**Comment by hoechenberger** [comment]
@dengemann Don't worry about this, I will address this when closing #401

**Comment by hoechenberger** [comment]
Merging, the doc build problem is something I will fix in a separate PR – it wasn't introduced here.

**Comment by hoechenberger** [comment]
Thank you, @dengemann!!

**Pull request merged**